### PR TITLE
update environment sdk version constraints

### PR DESCRIPTION
**Pull request by KevinTheGray**
0.1.6 is currently not compatible with the most recent versions of flutter, this is an update to fix that

Fixes https://github.com/flutter/flutter_markdown/issues/55

### CODE REVIEW

**Comment by KevinTheGray** [comment]
@mit-mit could I get a review?  thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.1.7
+
+* Updated environment sdk constraints
```
Comment (mit-mit) [inline CHANGELOG.md]: Should mention that this is to make it Dart 2 compatible, and that as of this change it required a Dart 2 SDK

**Comment by mit-mit** [comment]
Looks like tests pass now? I'm curious why we are not hitting https://github.com/flutter/flutter_markdown/issues/53 ?

**Comment by KevinTheGray** [comment]
Thanks @mit-mit, i'll update the changelog with that info shortly.  As for the  tests passing, I believe there may have been some updates to the `matcher` constraints in a recent version of flutter, but i'll double check in a bit

**Comment by KevinTheGray** [comment]
@mit-mit I made those changes to the changelog

For the tests passing, the fix was to update to a newer version of `mockito`.  However, doing so would cause a dependency constraint conflict in flutter, i.e., 
```
Running "flutter packages get" in testmockito...                 
Because every version of flutter_test from sdk depends on test 0.12.41 which depends on matcher >=0.12.2 <0.12.3, every version of flutter_test from sdk requires matcher >=0.12.2 <0.12.3.
And because mockito >=3.0.0-beta+2 depends on matcher ^0.12.3, flutter_test from sdk is incompatible with mockito >=3.0.0-beta+2.
So, because testmockito depends on both flutter_test any from sdk and mockito ^3.0.0-beta+2, version solving failed.
pub get failed (1)
```

This commit to flutter updated the `matcher` version and fixed the constraint issue: https://github.com/flutter/flutter/commit/686d8f8a22a8ad03375c0b7b8a2442831652190b#diff-6db519d06b539bb165a44ad589217e9e

and now the tests run properly.

**Comment by mit-mit** [comment]
sgtm, approved!

**Pull request merged**